packages/server/app/Controllers/Http/AuthenticationController.ts

```diff
@@ -355,6 +357,55 @@ export default class AuthenticationController {
         };
     }
 
+    public async resetPassword({
+        request,
+        auth,
+        response,
+    }: HttpContextContract): Promise<ResetPasswordResponseBody> {
+        const { email, password, token, authenticationMode } =
+            ResetPasswordRequestBody.parse(request.body());
+
+        const user = await User.findBy('email', email);
+        if (user === null) {
+            response.status(400);
+
+            return {
```

Comment:
question: why would this error be an invalid token ?

Reply:
Before this route is called, the email and the token pair must have been verified. I went to the simplest solution and decided that if email/token pair does not exist or is invalid, we return an `INVALID_TOKEN` error.

Reply:
It seems legit to me !